registry/auth/htpasswd/access.go

```diff
@@ -9,13 +9,17 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"sync"
 
 	"github.com/docker/distribution/context"
 	"github.com/docker/distribution/registry/auth"
 )
 
 type accessController struct {
 	realm    string
+	path     string
+	modtime  string
+	mu       *sync.Mutex
```

Comment:
I think the code is correct now, but please remove the `*` so this is simply `sync.Mutex`.